### DISCUSSION TITLE
AWS AMI Instance Types Supported

### DISCUSSION THREAD

**Discussion by SuiNom** [Security-Onion-Solutions/securityonion#6824] [Unsupported Versions] [answered]
Hi All,

After catching up on the SecurityOnion Conference I wanted to test out the honeypots functionality that was discussed [here](https://www.youtube.com/watch?v=NzUhfARVfJk&list=PLljFlTO9rB17mESq7Z9OeFKvVh39vJW34&index=4).

I've been using the terraform configuration found [here](https://github.com/Security-Onion-Solutions/securityonion-cloud) to deploy labs within AWS using the latest AMIs. I wanted to augment this with a honeypot but noticed that I couldn't get an AMI spun up on anything less than a `t3a.large` instance type, given the node will just be a honeypot, if feels slightly excessive (and expensive) to keep it with 2vCPUs and 8GB of RAM. I was hoping to get away with using a `t3a.micro` or `t3a.nano` instead.

Is there anyway to lower these AMI instance requirements, or am I better off spinning up a small CentOS base and install SO using the [documentation](https://github.com/Security-Onion-Solutions/securityonion-cloud)?

Thanks!

**Comment by weslambert** [comment, marked as answer]
You can experiment with different instance sizes.  We recommend and preset the type as such, as that is what we officially recommend/support. I'm not sure if a micro or nano would be possible (sounds like you are trying to reduce it to free/no cost?) However, If you are running a node with minimal services you might be able to lower the resources a bit. I'll let @defensivedepth add any additional comments around the honeypot node and it's requirements.

**Reply by defensivedepth** [reply]
So the way the honeypot node works is to convert an existing Forward node - which means the system needs to have enough resources to at least initially run the Forward node services. Once it is converted, the system requirements go way down.

I am looking at ways to make this easier to do.

**Reply by SuiNom** [reply]
Hey @defensivedepth ,

Great work on the honeypots, the presentation was awesome. I'm gonna to raise an issue on your repo shortly to document my install process and troubleshooting with the IDH Node and `SO 2.3.91` - I don't want to derail this thread though!

@weslambert, thanks for the answer. Rather than being able to tune the instance sizes when using the SecurityOnion2 AMI, the instance type seems to have a minimum requirement, trying to spin up an instance on anything lower than `t3a.large` hits an error:
```
aws_instance.honeypot2[0]: Creating...

Error: Error launching source instance: UnsupportedOperation: The instance configuration for this AWS Marketplace product is not supported. Please see the AWS Marketplace site for more information about supported instance types, regions, and operating systems.
	status code: 400, request id: e1fe6d66-dff8-4a79-a312-236667ada3fa

  on honeypot2.tf line 1, in resource "aws_instance" "honeypot2":
   1: resource "aws_instance" "honeypot2" {
```
The configuration that generated the error is below. But with my further testing, the failure makes sense as anything with less than 8GB RAM seemed to hit out of memory errors before completing the setup.
`honeypots2.tf`
```
variable "honeypots2" {
  default = 0
}

variable honeypot2_instance_type {
  type = string
  default = "t3a.micro"
}

resource "aws_instance" "honeypot2" {
  depends_on = [ aws_internet_gateway.default ]
  count         = var.honeypots2
  instance_type = var.honeypot2_instance_type
  ami           = data.aws_ami.latest_honeypot.id

  tags = {
    Name = "secret-db2-${count.index}"
    so-lab = "True"
  }

  subnet_id              = aws_subnet.default.id
  vpc_security_group_ids = [aws_security_group.securityonion.id]
  key_name               = aws_key_pair.auth.key_name
  private_ip             = "172.16.163.10${count.index}"

  root_block_device {
    delete_on_termination = true
    volume_size           = 30
  }
}

# Get latest Security Onion Image
data "aws_ami" "latest_honeypot" {
  
  most_recent = true
  owners = ["679593333241"]

  filter {
    name = "name"
    values = ["Security Onion ${var.soversion}*"]
  }
}
```
In the end I'm just looking for the flexibility to reduce the instance sizes where possible to reduce costs. I had a suspicion that locking the IDH Node at 8GB just to meet the AMI requirements would be a little wasteful, considering the normal sensor services should be stopped.

If anyone comes after, I had some success with the following:
1. Create a CentOS 7 `t3a.large` (8GB RAM or next step would fail with OOM errors)
2. Install SecurityOnion via the [documention page](https://docs.securityonion.net/en/2.3/installation.html#installation-on-ubuntu-or-centos)
3. Tuning the instance type after install - I went down to `t3a.micro` but I'm still hitting OOM errors, I think there's more that can be done there though, and I didn't have a completely working IDH setup at the end of it.

Thanks All!
